### PR TITLE
wifi: Bin patch loader

### PR DESCRIPTION
**Pull request by krish2718**
Removes C based patch headers support and instead adds support for binary based patch.

### CODE REVIEW

File: drivers/wifi/nrf700x/src/fw_load.c [hunk 1]
```diff
@@ -22,36 +21,205 @@
...
+	uint32_t type;
+	uint32_t len;
+	/* Data follows */
+} __packed;
+
+struct nrf70_fw_image_info {
+	uint32_t num_images;
+	uint32_t version;
+	uint32_t feature_flags;
+} __packed;
```
Comment (krish2718) [inline drivers/wifi/nrf700x/src/fw_load.c:98]: Will move this to `patch_info.h`

File: drivers/wifi/nrf700x/src/fw_load.c [hunk 2]
```diff
@@ -22,36 +21,205 @@
...
+
+static int nrf_wifi_patch_version_compat(const uint32_t version)
```
Comment (krish2718) [inline drivers/wifi/nrf700x/src/fw_load.c:100]: Will move this and other helpers to FMAC

File: drivers/wifi/nrf700x/src/fw_load.c [hunk 3]
```diff
@@ -22,36 +21,72 @@
...
-	memset(&fw_info, 0, sizeof(fw_info));
-	fw_info.lmac_patch_pri.data = (void *) nrf_wifi_lmac_patch_pri_bimg;
-	fw_info.lmac_patch_pri.size = sizeof(nrf_wifi_lmac_patch_pri_bimg);
-	fw_info.lmac_patch_sec.data = (void *) nrf_wifi_lmac_patch_sec_bin;
-	fw_info.lmac_patch_sec.size = sizeof(nrf_wifi_lmac_patch_sec_bin);
-	fw_info.umac_patch_pri.data = (void *) nrf_wifi_umac_patch_pri_bimg;
-	fw_info.umac_patch_pri.size = sizeof(nrf_wifi_umac_patch_pri_bimg);
-	fw_info.umac_patch_sec.data = (void *) nrf_wifi_umac_patch_sec_bin;
-	fw_info.umac_patch_sec.size = sizeof(nrf_wifi_umac_patch_sec_bin);
+	status = nrf_wifi_fmac_fw_parse(rpu_ctx, (const uint8_t *)_bin_nrf70_fw_start,
```
Comment (rado17) [inline drivers/wifi/nrf700x/src/fw_load.c:75]: Where are _bin_nrf70_fw_start and _bin_nrf70_fw_end defined?

Comment (krish2718) [reply drivers/wifi/nrf700x/src/fw_load.c:75]: They are defined by a simple `INCBIN` macro :) .

**Review by rado17 — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

**Review by VivekUppunda — APPROVED** [review]

**Pull request merged**